### PR TITLE
Pipeline: Fix flakiness on Pa11y CI 

### PR DESCRIPTION
**Pull request by axelavargas**
**Why we need this PR**:

- Pa11y-ci is broken due to a conflicting error with Uplot library, uplot introduced `??`, and for versions of pa11y < 6.0.x, that's a problem.
- Before the conflicting error pa11y-ci was not stable, it was giving non consistent number of errors, breaking the PRs.

**What this PR does**:
- Now Pa11y ci is pointing to commit where the new version of pa11y library has been update(6.0.x), we needed to do on this way, because the team from pa11y-ci is planning to do a release of v3.0, but we don't know when is going to happen.
- Added `wait:500` to each url in pa11y configs (pr and main), this will wait 500ms before performing the a11y test
- Added `useIncognitoBrowserContext: false,`, this will make sure we only need to login once.
- Improve `yarn:test:accessibility-report`, now it will instantiate locally the e2e instance, making sure we will have the same a11y errors as we see in drone. This command is only available on local environments.
- Thresholds were updated to be aligned with latest main changes.

**Special notes for your reviewer**:
I will leave the a11y step not mandatory in PR for now, I want to check how it behaves after 1 week with the new changes.

### CODE REVIEW

File: scripts/drone/steps/lib.star [hunk 1]
```diff
@@ -418,7 +418,7 @@ def test_a11y_frontend_step(edition, port=3001):
         },
         'failure': 'ignore',
         'commands': [
-            'yarn wait-on http://$HOST:$PORT',
+            'yarn wait-on http://$HOST:$PORT/login',
```
Comment (hugohaggmark) [inline scripts/drone/steps/lib.star:421]: Maybe we should have both lines? One waiting for the server and a second line for the actual page? Maybe each page needs to "warm-up" before we can run pa11y-ci?

Comment (axelavargas) [reply scripts/drone/steps/lib.star:421]: 🤔 yes, maybe having both lines will give more time, I also found a property called `wait`, is a time on milliseconds that we can pass to each URL before Pa11y is executed.

Comment (hugohaggmark) [reply scripts/drone/steps/lib.star:421]: The `wait` prop seems promising 👍

File: contribute/create-pull-request.md [hunk 2]
```diff
@@ -62,7 +62,7 @@ We use [pa11y-ci](https://github.com/pa11y/pa11y-ci) to collect accessibility er
 If the contribution introduces new a11y errors, our continuous integration will fail, preventing you to merge on the main branch. In those cases there are two alternatives for moving forward:
 
 - Check the error log on the pipeline step `test-a11y-frontend-pr`, identify what was the error, and fix it.
-- Locally run the command `yarn test:accessibility-report` that generates an HTML accessibility report, then go to the URL that contains your change, identify the error, and fix it. Keep in mind, a local Grafana instance needs to be running on `http://localhost:3000`.
+- Locally run the command `yarn test:accessibility-report` that generates an HTML accessibility report, then go to the URL that contains your change, identify the error, and fix it. Keep in mind, a local e2e Grafana instance needs is going to be running on `http://localhost:3001`.
```
Comment (thisisobate) [inline contribute/create-pull-request.md:65]: I guess we should remove 'needs' here?

File: .drone.yml [hunk 3]
```diff
@@ -150,6 +150,7 @@ steps:
   image: buildkite/puppeteer
   commands:
   - yarn wait-on http://$HOST:$PORT
+  - yarn wait-on http://$HOST:$PORT/login
```
Comment (dsotirakis) [inline .drone.yml:153]: Does this need to wait on both? Shouldn't this new line be enough?

**Review by dsotirakis — COMMENTED** [review]
So is this step going to spin up a Grafana instance and run against it? It's going to add some worth mentioning time on the (already) heavy pipeline of ours I think 😬

File: .drone.yml (hunk 3 above)
Comment (hugohaggmark) [reply .drone.yml:153]: I agree, I guess we don't need this wait any longer?

**Review by hugohaggmark — APPROVED** [review]
Looks great 🎉  

Now that I revisit this the first thing that strucks me is that `.pa11yci.conf.js` and `.pa11yci-pr.conf.js` look very similar or ~~perhaps identical~~ (thresholds are added to the PR one)? 

Do we need both or is this a separation of concerns thing?

File: package.json [hunk 4]
```diff
@@ -191,7 +191,7 @@
     "mutationobserver-shim": "0.3.3",
     "ngtemplate-loader": "2.1.0",
     "nodemon": "2.0.2",
-    "pa11y-ci": "2.4.2",
+    "pa11y-ci": "https://github.com/pa11y/pa11y-ci/tarball/5c842cf1b9fe2867b70ff5354851d985be8d71c4",
```
Comment (leeoniya) [inline package.json:194]:
```suggestion
"pa11y-ci": "pa11y/pa11y-ci#5c842cf1b9fe2867b70ff5354851d985be8d71c4",
```

**Comment by axelavargas** [comment]
> So is this step going to spin up a Grafana instance and run against it? It's going to add some worth mentioning time on the (already) heavy pipeline of ours I think

Hallo 👋🏾, not really, on Drone we are using the existing e2e grafana instance (_the same one we were using for the e2e test step_).  

In the PR I mentioned: 

> Improve yarn:test:accessibility-report, now it will instantiate locally the e2e instance, making sure we will have the same a11y errors as we see in drone.

But I forgot to add that this command is only available on local environments, for debugging purposes.

File: .drone.yml (hunk 3 above)
Comment (axelavargas) [reply .drone.yml:153]: that's true, I don't think we need it anymore :)

**Comment by axelavargas** [comment]
> Looks great 
> 
> Now that I revisit this the first thing that strucks me is that `.pa11yci.conf.js` and `.pa11yci-pr.conf.js` look very similar or ~perhaps identical~ (thresholds are added to the PR one)?
> 
> Do we need both or is this a separation of concerns thing?

Thank you @hugohaggmark for checking the PR 🙌🏾, and indeed, the reason behind is because we need to add thresholds, unfortunately, pa11y does not give us a report with all the errors when thresholds are passed, therefore we can't send the a11y number of errors to the Grafana CI Code Dashboard. 

There is an [issue](https://github.com/pa11y/pa11y-ci/issues/70) open from the pa11y community, maybe in the future it will added 🤞🏾

**Comment by hugohaggmark** [comment]
> > Looks great
> > Now that I revisit this the first thing that strucks me is that `.pa11yci.conf.js` and `.pa11yci-pr.conf.js` look very similar or ~perhaps identical~ (thresholds are added to the PR one)?
> > Do we need both or is this a separation of concerns thing?
> 
> Thank you @hugohaggmark for checking the PR 🙌🏾, and indeed, the reason behind is because we need to add thresholds, unfortunately, pa11y does not give us a report with all the errors when thresholds are passed, therefore we can't send the a11y number of errors to the Grafana CI Code Dashboard.
> 
> There is an [issue](https://github.com/pa11y/pa11y-ci/issues/70) open from the pa11y community, maybe in the future it will added 🤞🏾

Great! Maybe we could common parts into a separate file and require/import it? Might be overkill 🙈

**Comment by dsotirakis** [comment]
> > So is this step going to spin up a Grafana instance and run against it? It's going to add some worth mentioning time on the (already) heavy pipeline of ours I think
> 
> Hallo 👋🏾, not really, on Drone we are using the existing e2e grafana instance (_the same one we were using for the e2e test step_).
> 
> In the PR I mentioned:
> 
> > Improve yarn:test:accessibility-report, now it will instantiate locally the e2e instance, making sure we will have the same a11y errors as we see in drone.
> 
> But I forgot to add that this command is only available on local environments, for debugging purposes.

Good, that does make sense then 👏 Nice work!!!

**Comment by axelavargas** [comment]
Ups, I made a mistake backporting this to 8.1.x, it does not need to be there 😅, the PR only involves CI changes

**Pull request merged**